scripts/up.sh

```diff
@@ -1,79 +1,54 @@
-PARALLEL="--parallel"
+ENV_FILE=".env"
+echo "up.sh: sourcing environment vars from-->$ENV_FILE"
+source "$ENV_FILE"
 
-show_help() {
-  echo "Manage bringing up the SDK test environment."
-  echo
-  echo "Usage: up.sh [options]"
-  echo
-  echo "Options:"
-  echo "  -f <FILE>  Override the environment file."
-  echo "  -t <TYPE>  Override the installation type specified in the environment file."
-  echo "             Valid types: ['channel', 'type']"
-  echo "  -s         Skip rebuilding the docker image."
-  echo "  -i         Start the docker environment in interactive mode."
-  echo "  -h         Provide this help information."
-  echo "  -p         Disable parallel build mode. Probably only useful when editing the test environment."
-}
-
-# Parse arguments
-while getopts "pf:t:sih" opt; do
-  case "$opt" in
-    f) ENV_FILE=$OPTARG; ;;
-    i) unset DAEMON_FLAG; ;;
-    s) SKIP_BUILD=1; ;;
-    t) TYPE_OVERRIDE=$OPTARG; ;;
-    h) show_help; exit 0 ;;
-    p) unset PARALLEL; ;; 
-  esac
-done
-
-# Verify there are no positional parameters with getopt/getopts
-shift "$((OPTIND-1))"
-if [[ "$1" != "" ]]; then
-  echo "No positional arguments should be provided, found '$@'"
-  echo
-  show_help
-  exit
-fi
-
-# Load environment.
-source $ENV_FILE
-
-# Choose which dockerfile to use.
-TYPE=${TYPE_OVERRIDE:-$TYPE}
 if [[ $TYPE == "channel" ]] || [[ $TYPE == "source" ]]; then
   export TYPE="$TYPE"
+  if [[ $TYPE == "channel" ]]; then
+    ALGOD_URL=""
+    ALGOD_BRANCH=""
+    ALGOD_SHA=""
+  else
+    ALGOD_CHANNEL=""
+  fi
 else
   echo "Unknown environment: $TYPE"
   exit 1
 fi
 
-echo "Bringing up network with '$TYPE' configuration."
-
-# Make sure it isn't running
+# Make sure test-sdk sandbox isn't running and clean up any docker detritous
+echo "up.sh: Before bootstrapping, try cleaning up first..."
 ./scripts/down.sh
+rm -rf "$LOCAL_SANDBOX_DIR"
+echo "up.sh: seconds it took to get to end of $LOCAL_SANDBOX_DIR cleanup: " + $(($(date "+%s") - $START))
 
-# Remove the containers to allow re-running stateful tests.
-docker-compose rm --force
 
-# When developing, it's often useful to skip the build phase.
-if [[ $SKIP_BUILD -eq 0 ]]; then
-  docker-compose build --no-cache $PARALLEL
-fi
+rootdir=$(dirname "$0")
+pushd "$rootdir"/.. > /dev/null || exit
+
+git clone --branch "$SANDBOX_BRANCH" --single-branch "$SANDBOX_URL" "$LOCAL_SANDBOX_DIR"
+
+cp .env "$LOCAL_SANDBOX_DIR"/.
+
+SANDBOX_CFG="config.harness"
+echo "up.sh: about to envsubst < $SANDBOX_CFG  > $LOCAL_SANDBOX_DIR/$SANDBOX_CFG"
+envsubst < "$SANDBOX_CFG" > "$LOCAL_SANDBOX_DIR/$SANDBOX_CFG"
+
+echo "up.sh: resulting $LOCAL_SANDBOX_DIR/$SANDBOX_CFG:"
+cat "$LOCAL_SANDBOX_DIR/$SANDBOX_CFG"
+
+echo ""
+echo "up.sh: seconds it took to get to end of cloning sandbox into $LOCAL_SANDBOX_DIR: " + $(($(date "+%s") - $START))
+echo ""
+echo "up.sh: bringing up network with TYPE=$TYPE configuration."
+
+pushd "$LOCAL_SANDBOX_DIR"
 
-docker-compose up $DAEMON_FLAG
+./sandbox up -v harness
```

Comment:
@tzaffi Optional:  Can you either remove (preferred - less effort) or make `-v` a user-configurable flag (default to off)?

* During my local tests, verbose mode output swamps all other output.  I think the user is susceptible to missing sandbox-related output.
* If wiring through a flag is more effort than you prefer, then I think it's reasonable to omit the flag entirely.  A user can make local edits to enable the flag (as I did to disable it).
* If you strongly prefer as is, then I'm good to resolve as is.  I don't feel strongly enough to hold the PR.

Reply:
I believe it's easy enough to wire it through and default to silent. I'll keep this issue open until I'm clear of the outcome.

Reply:
I'm making this configurable by introducing a `VERBOSE_HARNESS` variable in `.env`